store/sqlstore/system_store.go

```diff
@@ -130,18 +136,15 @@ func (s SqlSystemStore) PermanentDeleteByName(name string) (*model.System, error
 // InsertIfExists inserts a given system value if it does not already exist. If a value
 // already exists, it returns the old one, else returns the new one.
 func (s SqlSystemStore) InsertIfExists(system *model.System) (*model.System, error) {
-	tx, err := s.GetMaster().BeginTx(context.Background(), &sql.TxOptions{
-		Isolation: sql.LevelSerializable,
```

Comment:
We need to preserve the isolation level here. I think it might be okay to add a new method on `*sqlxDBWrapper` called `BeginXWithIsolation` that allows us to set an isolation level. Or if you want to use raw methods for this, that's okay as well.